libnetdata/threads/threads.c

```diff
@@ -121,6 +121,26 @@ static void *thread_start(void *ptr) {
     if(pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL) != 0)
         error("cannot set pthread cancel state to ENABLE.");
 
+    if (netdata_thread->tag) {
+        char *threadname = strdupz(netdata_thread->tag);
+        if (threadname) {
+            // Name if limited to 16 chars
+            threadname[15] = 0;
+#if defined(__FreeBSD__)
+            if (pthread_set_name_np(pthread_self(), threadname) != 0)
+                error("cannot set pthread name of %d to %s", gettid(), threadname);
+#elif defined(__APPLE__)
+            if (pthread_setname_np(threadname) != 0)
+                error("cannot set pthread name of %d to %s", gettid(), threadname);
+#else
+            if (pthread_setname_np(pthread_self(), threadname) != 0)
+                error("cannot set pthread name of %d to %s", gettid(), threadname);
+#endif
+            info("set name of thread %d to %s", gettid(), threadname);
+            free(threadname);
+        }
+    }
```

Comment:
I think it would be nice to move set tag stuff to a separate func. @mfundul

Reply:
It would be nice I agree, I did not specifically ask it cause it's only one place for now, but I also thought of it.

Reply:
Created dedicated static func `thread_set_name` + added return code in log message if error.